### DISCUSSION TITLE
Konsist not detecting annotations in our tests.

### DISCUSSION THREAD

**Discussion by Carrieukie** [LemonAppDev/konsist#818] [Report a Problem]
### Tell us what version of Konsist you use.

v0.13.0

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

We have written Konsists test to make sure certain annotations exist on Composable screens. It is working for some but it's not for others.

Here's the test.
    @Test
    fun `functions with 'Screen' suffix should have 'Destination' and 'Composable' annotation`() {
        Konsist.scopeFromProject()
            .functions()
            .withNameEndingWith("Screen")
            .assertTrue {
                it.hasAnnotationsWithAllNames("Destination", "Composable")
            }
    }

And here's the class it's failing on.
@Destination
@Composable
@ExperimentalComposeUiApi
@ExperimentalFoundationApi
@ExperimentalMaterialApi
fun ProductCatalogueScreen(
    navigator: MarketNavigator,
    productCatalogueViewModel: ProductCatalogueViewModel = hiltViewModel(),
) {}

### Actual behaviour

com.lemonappdev.konsist.core.exception.KoAssertionFailedException: Assert 'functions with 'Screen' suffix should have 'Destination' and 'Composable' annotation' was violated (1 time). Invalid declarations:
/Users/karis/StudioProjects/customerapp/market/presentation/src/main/java/catalogue/ProductCatalogueScreen.kt:792:1 (OrdersScreen FunctionDeclaration)

### Expected behaviour

The test is expected to pass.

**Comment by igorwojda** [comment]
@Carrieukie we are unable to reproduce this. 
Please use backticks to format the code or attach a file.

**Comment by igorwojda** [comment]
@Carrieukie can you give it a try or refromat these snippets, so we cna reproduce it?

**Reply by Carrieukie** [reply]
Hey @igorwojda it turned out to be a package structure issue on our side, but we've resolved it now. Apologies for not updating you on this earlier!

**Comment by igorwojda** [comment]
Do you remember the details? Can you share some light on a concrete example - perhaps Konsist could detect and handle edge case ?

**Reply by superus8r** [reply]
I had a similar issue when using Konsist to check occurrence of `@Composable` annotations and were able to solve it.
My mistake was to use `files` instead of `functions()`.

This wouldn't work:
```
Konsist.scopeFromProject().files.
        withAnnotationOf(Composable::class)
``` 
I believe it is because Konsist looks for Annotations only in Class declarations by default, but composable are usually defined as functions in files without classes.

Telling Konsist to explicitly look for the annotations for functions worked:
```
Konsist.scopeFromProject().functions().
        withAnnotationOf(Composable::class)
```
